cfgov/v1/tests/atomic_elements/organisms/test_contact_expandables.py

```diff
@@ -0,0 +1,138 @@
+import json
+
+from django.http import HttpRequest
+from django.test import TestCase
+
+from wagtail.tests.utils import WagtailTestUtils
+from wagtail.wagtailcore import blocks
+
+from v1.atomic_elements.organisms import (
+    ContactExpandable, ContactExpandableGroup
+)
+from v1.models import Contact
+
+
+class ContactExpandableTests(TestCase):
+    def test_render(self):
+        contact = Contact.objects.create(
+            heading="Scrooge McDuck",
+            body="World's richest duck",
+            contact_info=json.dumps([
+                {
+                    'type': 'hyperlink',
+                    'value': {
+                        'url': 'https://vault.duckburg.com',
+                        'text': 'Money Bin',
+                    },
+                },
+            ])
+        )
+
+        block = ContactExpandable()
+        html = block.render(block.to_python({'contact': contact.pk}))
+
+        self.assertIn("Scrooge McDuck", html)
+        self.assertNotIn("World's richest duck", html)
+        self.assertInHTML(
+            '<a href="https://vault.duckburg.com">Money Bin</a>',
+            html
+        )
+
+    def test_bulk_to_python(self):
+        Contact.objects.bulk_create(
+            Contact(pk=i, heading=str(i))
+            for i in range(3)
+        )
+
+        class TestStreamBlock(blocks.StreamBlock):
+            contact = ContactExpandable()
+
+        block = TestStreamBlock()
+        value = blocks.StreamValue(
+            block,
+            [
+                {'type': 'contact', 'value': {'contact': 0}},
+                {'type': 'contact', 'value': {'contact': 1}},
+                {'type': 'contact', 'value': {'contact': 2}},
+            ],
+            is_lazy=True
+        )
+
+        # This verifies that the block does a single database query to retrieve
+        # all 3 Contacts, even though they are split up between three different
+        # ContactExpandable blocks.
+        with self.assertNumQueries(1):
+            block.render(value)
+
+
+class ContactExpandableGroupTests(WagtailTestUtils, TestCase):
+    def setUp(self):
+        Contact.objects.bulk_create(
+            Contact(
+                pk=i,
+                heading=str(i),
+                contact_info=json.dumps([
+                    {
+                        'type': 'hyperlink',
+                        'value': {
+                            'url': 'https://%d.example.com' % i,
+                        },
+                    },
+                ])
+            ) for i in range(10)
+        )
+
+    def test_render(self):
+        block = ContactExpandableGroup()
+        html = block.render(block.to_python({
+            'heading': "Contacts",
+            'expandables': [
+                {'contact': pk}
+                for pk in (5, 3, 2, 1, 8)
+            ],
+        }))
+
+        self.assertInHTML("<h3>Contacts</h3>", html)
+        self.assertNotIn("World's richest duck", html)
```

Comment:
Is this assertion meaningful? I might have missed something, but I didn't think this string even appears in any of the `Contact` objects that exist for this test.

Reply:
You're right, this was another copy/paste mistake, as this is used in the test above. I've added a test message there to clarify the purpose of this line, and removed it here.